### PR TITLE
Image commit on start

### LINKED ISSUE #584: False-positve on image removal from the staging
I have seen the missing images in several comments. This is a rare event but still

I think, there are two reasons for this:

1. sometimes cleanup activated on exact TTL and commit may be delayed for at least 10ms
1. the logic of submitting multiple ids is sequential and blocking. I didn't expect it to delay commit significantly and still don't understand how this even possible, but I see an example of a failing attempt to commit already removed staging image with a strangely long delay between cleanup (07:46:34.384) and commit (07:47:22.677)

### PR DESCRIPTION
**Pull request by paskal**
Resolves #584. Relies on merging #692 first.

## Behavior before this PR
1. Set `EDIT_TIME=20s` in docker-compose.yml
2. Start dev instance using docker-compose, post comment with image
3. Restart docker-compose
4. Wait for 2 minutes
5. Reload the page: image from the comment is no longer available as it was stuck in Staging without commit and then was deleted after `EDIT_TIME*5`, `1m40s` in this case.

## Behavior after this PR
1. Set `EDIT_TIME=20s` in docker-compose.yml
2. Start dev instance using docker-compose, post comment with image
3. Restart docker-compose
4. Wait for 2 minutes
5. Reload the page: image is still in place

### CODE REVIEW

File: backend/app/store/service/service.go [hunk 1]
```diff
@@ -200,6 +200,30 @@ func (s *DataStore) DeleteUserDetail(siteID string, userID string, detail engine
...
+		return err
+	}
+	ts := info.FirstStagingImageTS
+	if ts.IsZero() {
+		return nil
+	}
+	for _, site := range sites {
+		locator := store.Locator{SiteID: site}
+		comments, err := s.FindSince(locator, "time", store.User{}, ts)
+		if err != nil {
```
Comment (umputun) [inline backend/app/store/service/service.go:217]: as this operation runs for different sites it makes sense not to terminate on the first error but collect in multierror and continute.And return collected error(s) after this

File: backend/app/store/service/service_test.go [hunk 2]
```diff
@@ -1300,6 +1302,88 @@ func TestService_submitImages(t *testing.T) {
...
+	_, err := b.Engine.Create(comment)
+	require.NoError(t, err)
+
+	// resubmit single comment with single image
+	mockStore.On("Info").Once().Return(image.StoreInfo{FirstStagingImageTS: time.Time{}.Add(time.Second)}, nil)
+	err = b.ResubmitStagingImages([]string{"radio-t"})
+	assert.NoError(t, err)
+
+	// wait for Submit goroutine to commit image
+	mockStore.On("Commit", "dev_user/bqf122eq9r8ad657n3ng").Once().Return(nil)
```
Comment (umputun) [inline backend/app/store/service/service_test.go:1336]: the test will be more convincing if at least 2 images processed

File: backend/app/store/service/service_test.go [hunk 3]
```diff
@@ -1300,6 +1302,88 @@ func TestService_submitImages(t *testing.T) {
...
+
+	// resubmit single comment with single image
+	mockStore.On("Info").Once().Return(image.StoreInfo{FirstStagingImageTS: time.Time{}}, nil)
+	err := b.ResubmitStagingImages([]string{"radio-t"})
+	assert.NoError(t, err)
+
+	mockStore.AssertNumberOfCalls(t, "Info", 1)
+}
+
+func TestService_ResubmitStagingImagesError(t *testing.T) {
```
Comment (umputun) [inline backend/app/store/service/service_test.go:1365]: will be nice to cover the case with partial failure (one passed, one failed)

File: backend/app/store/image/image.go [hunk 4]
```diff
@@ -76,6 +88,10 @@ type submitReq struct {
 
 // NewService returns new Service instance
 func NewService(s Store, p ServiceParams) *Service {
+	p.commitTTL = p.EditDuration * 15 / 10  // Commit call on every 1.5 * EditDuration
```
Comment (umputun) [inline backend/app/store/image/image.go:91]: how 15/10 is easier to understand? even comment saying 1.5x

File: backend/app/store/image/image.go [hunk 5]
```diff
@@ -76,6 +88,10 @@ type submitReq struct {
...
+	p.cleanupTTL = p.EditDuration * 25 / 10 // Cleanup call on every 2.5 * EditDuration
```
Comment (umputun) [inline backend/app/store/image/image.go:92]: the same, this is 2.5x and easer to read as `*2.5`

File: backend/app/store/image/image.go [hunk 6]
```diff
@@ -133,23 +149,28 @@ func (s *Service) ExtractPictures(commentHTML string) (ids []string, err error)
...
 
 	for {
 		select {
 		case <-ctx.Done():
 			log.Printf("[INFO] cleanup terminated, %v", ctx.Err())
 			return
-		case <-time.After(s.TTL / 2): // cleanup call on every 1/2 TTL
-			if err := s.store.Cleanup(ctx, s.TTL); err != nil {
+		case <-time.After(s.cleanupTTL):
+			if err := s.store.Cleanup(ctx, s.EditDuration); err != nil {
```
Comment (umputun) [inline backend/app/store/image/image.go:162]: the knowlege about internalas of other parts with such a concete EditDuration name looks strange. Not sure what should be the right name.Probably this is cleanupTTL, and the thing you currently call cleanupTTL is some sort of clanupInterval or cleanupDuration.

File: backend/app/store/image/fs_store.go [hunk 7]
```diff
@@ -115,6 +115,33 @@ func (f *FileSystem) Cleanup(_ context.Context, ttl time.Duration) error {
 	return errors.Wrap(err, "failed to cleanup images")
 }
 
+// Info returns meta information about storage
+func (f *FileSystem) Info() (StoreInfo, error) {
+	if _, err := os.Stat(f.Staging); os.IsNotExist(err) {
+		return StoreInfo{}, nil
```
Comment (umputun) [inline backend/app/store/image/fs_store.go:121]: this is top-level API func and returned errors better to be wrapped with some context

File: backend/app/store/image/image.go (hunk 4 above)
Comment (paskal) [reply backend/app/store/image/image.go:91]: this has to be integers and not float value, I can't put `* 1.5` there unfortunately.

File: backend/app/store/image/image.go (hunk 6 above)
Comment (paskal) [reply backend/app/store/image/image.go]: Changed to `cleanupTTL` which is more in-line with original value.

File: backend/app/store/image/image.go (hunk 4 above)
Comment (umputun) [reply backend/app/store/image/image.go:91]: ah, ok.

**Pull request merged**